src/libData/AccountData/Account.cpp

```diff
@@ -190,11 +343,37 @@ Address Account::GetAddressFromPublicKey(const PubKey& pubKey)
     return address;
 }
 
-void Account::SetCode(std::vector<unsigned char>&& code)
+Address Account::GetAddressForContract(const Address& sender,
+                                       const uint256_t& nonce)
 {
-    m_codeCache = std::move(code);
+    Address address;
+
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;
+    sha2.Update(sender.asBytes());
+    vector<unsigned char> nonceBytes;
+    SetNumber<uint256_t>(nonceBytes, 0, nonce, UINT256_SIZE);
+    sha2.Update(nonceBytes);
+
+    const vector<unsigned char>& output = sha2.Finalize();
+    assert(output.size() == 32);
+
+    copy(output.end() - ACC_ADDR_SIZE, output.end(), address.asArray().begin());
```

Comment:
I've seen this many times in our codebase and maybe it looks common and becomes an acceptable way but actually, this bypasses the constructor of `Address` by setting the data directly using its API.